data/scripts.js

```diff
@@ -1170,6 +1170,7 @@ let BattleScripts = {
 		// {gigantamax?: string, maxMoves: {[k: string]: string} | null}[]
 		if (!skipChecks) {
 			if (!pokemon.canDynamax) return;
+			if (pokemon.transformed && (pokemon.template.baseSpecies === 'zacian' || pokemon.template.baseSpecies === 'zamazenta' || pokemon.template.baseSpecies === 'eternatus')) return;
```

Comment:
This doesn't do anything because the base species is a name, not an ID. (But by my reading of the following code, Ditto transformed into those legendaries should already be blocked from Dynamaxing.)

Reply:
I did tests on this before and after, before this ditto was able to dynamax if it transformed into those legendaries, and after with these changes, and ditto was not able to